torchvision/prototype/models/_utils.py

```diff
@@ -1,13 +1,123 @@
+import functools
+import inspect
 import warnings
-from typing import Any, Dict, Optional, TypeVar
+from typing import Any, Dict, Optional, TypeVar, Callable, Tuple, Union, cast
+from warnings import warn
 
-from ._api import Weights
+from torch import nn
+from torchvision.prototype.utils._internal import sequence_to_str
 
+from ._api import Weights
 
 W = TypeVar("W", bound=Weights)
+M = TypeVar("M", bound=nn.Module)
 V = TypeVar("V")
 
 
+def handle_positional_to_keyword_only(
```

Comment:
I would suggest to use https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/utils/validation.py#L36 instead. It's been extensively used in scikit-learn and it's just a decorator, no need to pass anything

Reply:
Yes, this is what I meant in https://github.com/pytorch/vision/pull/5001#issuecomment-982412063. I've added a general decorator in the latest commit.